internal/cmd/util/exit.go

```diff
@@ -23,10 +23,24 @@ import (
 
 // SetExitCode sets the exit code to 1 if the error is not a context.Canceled error.
 func SetExitCode(err error) {
-	if (err != nil) && errors.Is(err, context.DeadlineExceeded) {
-		errorExitCodeChannel <- 124 // Indicate that there was a timeout error
-	} else if (err != nil) && !errors.Is(err, context.Canceled) {
-		errorExitCodeChannel <- 1 // Indicate that there was an error
+	switch {
+	case err == nil || errors.Is(err, context.Canceled):
+		// If the context was canceled, we don't need to set the exit code
+	case errors.Is(err, context.DeadlineExceeded):
+		SetExitCodeValue(124) // Indicate that there was a timeout error
+	default:
+		SetExitCodeValue(1) // Indicate that there was an error
 	}
-	// If the context was canceled, we don't need to set the exit code
+}
+
+// SetExitCode sets the exit code to 1 if the error is not a context.Canceled error.
+func SetExitCodeValue(code int) {
```

Comment:
This new function was introduced so we can explicitly set an exit code (without relying on the error matching logic above).
This is useful when we are overwriting the `cmdutil.BehaviorOnFatal` which itself already provides us with the exit code that should be set.

Additionally, this function now allows being called twice, only the first call will be used to register the correct exit code. This is useful in case you have a very specific error code and call `SetExitCodeValue()`, after which you return an error and another error handling step tries to set the exit code to a less-specific value (eg. 1 for error).